mlflow/models/evaluation/base.py

```diff
@@ -1539,12 +1562,42 @@ def pred_sample(eval_df, _builtin_metrics, _artifacts_dir):
         )
     elif isinstance(model, PyFuncModel):
         pass
+    elif model is None:
+        # Evaluating a static dataset
+        if prediction is None:
+            raise MlflowException(
+                message="The prediction argument must be specified when model=None.",
+                error_code=INVALID_PARAMETER_VALUE,
+            )
+
+        if not isinstance(data, pd.DataFrame):
+            raise MlflowException(
+                message="The data must be a pandas dataframe when model=None.",
+                error_code=INVALID_PARAMETER_VALUE,
+            )
+
+        if prediction not in data.columns:
+            raise MlflowException(
+                message=f"The prediction column {prediction} is not found in data.",
+                error_code=INVALID_PARAMETER_VALUE,
+            )
+
+        data_with_prediction = data.copy(deep=True)
+        data = data.drop(columns=[prediction])
+
+        def fn(model_input: pd.DataFrame):
+            filtered_df = pd.merge(
+                model_input, data_with_prediction, how="inner", on=model_input.columns.tolist()
+            )
+            return filtered_df[prediction].tolist()
+
+        model = _get_model_from_function(fn)
```

Comment:
I don't think this would work for cases where the model_type is question_answering or text where we pass in one model_input at a time. If its one row at a time returning back the entire list wouldn't work.

Can we add a test case for model_type question_answering or text?

Reply:
`question_answering` is working. Added a test for it.